core/fetcher/fetcher.go

```diff
@@ -230,7 +230,7 @@ func (f *Fetcher) fetchProposerData(ctx context.Context, slot int64, defSet core
 			return nil, err
 		}
 
-		randao := randaoData.Signature().ToETH2()
+		randao := randaoData.Signatures()[0].ToETH2()
```

Comment:
Thougth: can we ever unmarshal a `SignedData` whose signatures slice is empty or nil?

Reply:
Yes, such `SignedData` can be unmarshalled. In general, marshalling/unmarshalling is not related to the signatures.

Take for example the `Attestation` object:
```go
func (a Attestation) Signatures() []Signature {
	return []Signature{SigFromETH2(a.Attestation.Signature)}
}
```
```go
func (a *Attestation) UnmarshalJSON(b []byte) error {
	return a.Attestation.UnmarshalJSON(b)
}
```
Unmarshalling can fail, but not because of nil signatures as these two are not directly co-related

Reply:
Then maybe we should add bound checks?

Reply:
I don't think we need to add bound checks, as the unmarshalling is done by the go-eth2-client itself and it does bound checks